src/main/java/net/imagej/ops/topology/EulerCharacteristic26N.java

```diff
@@ -0,0 +1,112 @@
+package net.imagej.ops.topology;
+
+import net.imagej.ops.Contingent;
+import net.imagej.ops.Op;
+import net.imagej.ops.special.function.AbstractUnaryFunctionOp;
+import net.imglib2.RandomAccess;
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.type.BooleanType;
+import net.imglib2.view.Views;
+import org.scijava.plugin.Plugin;
+
+/**
+ * An Op which calculates the euler characteristic (χ) of the given 3D binary image.
+ * Here Euler characteristic is defined as χ = β_0 - β_1 + β_2, where β_i are so called Betti numbers.
+ * β_0 = number of separate particles
+ * β_1 = number of handles
+ * β_2 = number enclosed cavities
+ * <p>
+ * The Op calculates χ by using the triangulation algorithm described by Toriwaki & Yonekura (see below).
+ * There it's calculated X = ∑Δχ(V), where V is a 2x2x2 neighborhood around each point in the 3D space.
+ * We are using the 26-neighborhood version of the algorithm. The Δχ(V) values here are predetermined.
+ * <p>
+ * For the algorithm see
+ * Toriwaki J, Yonekura T (2002) Euler Number and Connectivity Indexes of a Three Dimensional Digital Picture.
+ * Forma 17: 183-209.
+ * <a href="http://www.scipress.org/journals/forma/abstract/1703/17030183.html">
+ * http://www.scipress.org/journals/forma/abstract/1703/17030183.html</a>
+ * <p>
+ * For the Betti number definition of Euler characteristic see
+ * Odgaard A, Gundersen HJG (1993) Quantification of connectivity in cancellous bone,
+ * with special emphasis on 3-D reconstructions.
+ * Bone 14: 173-182.
+ * <a href="http://dx.doi.org/10.1016/8756-3282(93)90245-6">doi:10.1016/8756-3282(93)90245-6</a>
+ *
+ * @author Richard Domander
+ * @author David Legland (original MatLab implementation)
+ */
+@Plugin(type = Op.class, name = "eulerCharacteristic26N")
+public class EulerCharacteristic26N<B extends BooleanType>
+        extends AbstractUnaryFunctionOp<RandomAccessibleInterval<B>, Double> implements Contingent {
```

Comment:
We should consider carefully what return type to use.

`Double` is immutable, and therefore unsuitable for use with `ComputerOp`s.

If you look at the stats ops such as stats.mean (e.g., `DefaultMean`), you'll see that they are hybrid ops which produce an `O extends RealType<O>`. Unfortunately, this actually causes a problem with generics. Really the best structure is to use two ops:

* `DefaultMeanFunction<I extends RealType<I>, O extends RealType<O>> extends UFViaUC<I, DoubleType, I, O>`
* `DefaultMeanComputer<I extends RealType<I>, O extends RealType<O>> implements AbstractUnaryComputerOp<I, O>`

That way, you end up with two method signatures:

* `<I extends RealType<I>> DoubleType mean(I in)`
* `<I extends RealType<I>, O extends RealType<O>> O mean(I in, O out)`

The `UFViaUC` is a "unary function via unary computer" convenience class where the function computation delegates to a computer. However, if you study the generics above carefully, they are still a bit funny. I may need to rethink this.

Anyway, I think we want to follow a similar pattern here. Of course, I am open to any ideas to make this pattern as painless as possible. (In particular, it sucks needing to create the function wrapper version of such ops every time.)